### PR TITLE
fix: empty build due babel incorrect configuration

### PR DESCRIPTION
**Pull request by ghassanmas**
The PR #315 added @babel/core >= 7.*. upgrading/using babel v7
  requires, making certian changes, e.g using @babel...etc. This
  lead to `npm run build` being interrupted/exited.without
  generating the /dist folder.

  Luckily, there is a pacakge to automate migration of config
  using `npx babel-upgrade --write`, followed by `npm install`, and
  few minor changes, the dist folder came to life again.

  This shall fixes this openedx/build-test-release-wg#166. and
  probably making this openedx/edx-platform#30309 unnecessary.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ghassanmas! I've created [OSPR-6648](https://openedx.atlassian.net/browse/OSPR-6648) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by ghassanmas** [comment]
@Jawayria @mraarif since this is realted to #340 I am tagging you here.

**Comment by mraarif** [comment]
@Jawayria can you help @ghassanmas

File: package.json [hunk 1]
```diff
@@ -39,20 +39,21 @@
...
+    "@babel/preset-env": "^7.0.0",
+    "@babel/preset-react": "^7.0.0",
     "@commitlint/cli": "^6.1.3",
     "@commitlint/config-angular": "^6.1.3",
     "@edx/reactifex": "^1.1.0",
     "autoprefixer": "^7.2.5",
-    "babel-cli": "^6.26.0",
-    "babel-eslint": "^7.2.3",
-    "babel-jest": "^21.2.0",
+    "babel-core": "^7.0.0-bridge.0",
```
Comment (Jawayria) [inline package.json:52]: We already have `@babel/core` in dependencies then why do we need to add `babel-core`?

Comment (ghassanmas) [reply package.json]: Yup good catch just removed it. May be was added by the script or by me while debugging and forgot to remove it... Anyway after removing still got the bundle when `npm run build`

**Review by Jawayria — APPROVED** [review]

**Comment by aht007** [comment]
@ghassanmas I enabled CI checks on this PR and looks like some configuration needs updates that you might have missed after replacing the outdated packages with newer packages

**Comment by ghassanmas** [comment]
The issue here is because we upgrade babel to v7 while using still using `babel-jest` 23.x. 
As it seems `babel-jest` 23.* uses babel 6 and where we upgrade to babel 7, this means we have to rely to on https://github.com/babel/babel-bridge, which is the official recommend solution to get away when you have pacakges that needs babel 6 and babel 7.  Also qouting from babel-jest: 
> Note: If you are using Babel version 7 then you need to install babel-jest, babel-core@^7.0.0-bridge.0 and @babel/core with the following command:
yarn add --dev babel-jest babel-core@^7.0.0-bridge.0 @babel/core regenerator-runtime
You will need to use babel.config.js in order to transpile node_modules. See https://babeljs.io/docs/en/next/config-files for more information.
You can also see the example in the Jest repository: https://github.com/facebook/jest/tree/54f4d4ebd3d1a11d65962169f493ce41efdd784f/examples/babel-7 [^1]

So resolve I would need to revert this commit [8a14cdd](https://github.com/openedx/studio-frontend/pull/341/commits/8a14cdd55f5eb8ee89223c78e86791ce09b3b6ea). And that pacakge shall be removed when `babel-jest` is upgraded to >= 24.*
Does that sound okay @Jawayria @aht007 

[^1]: https://archive.jestjs.io/docs/en/23.x/getting-started.html#using-babel

**Comment by aht007** [comment]
> The issue here is because we upgrade babel to v7 while using still using `babel-jest` 23.x. As it seems `babel-jest` 23.* uses babel 6 and where we upgrade to babel 7, this means we have to rely to on https://github.com/babel/babel-bridge, which is the official recommend solution to get away when you have pacakges that needs babel 6 and babel 7. Also qouting from babel-jest:
> 
> > Note: If you are using Babel version 7 then you need to install babel-jest, babel-core@^7.0.0-bridge.0 and @babel/core with the following command:
> > yarn add --dev babel-jest babel-core@^7.0.0-bridge.0 @babel/core regenerator-runtime
> > You will need to use babel.config.js in order to transpile node_modules. See https://babeljs.io/docs/en/next/config-files for more information.
> > You can also see the example in the Jest repository: https://github.com/facebook/jest/tree/54f4d4ebd3d1a11d65962169f493ce41efdd784f/examples/babel-7 [1](#user-content-fn-1-e2b1bc6da4c6808b4eb6bb5149b517ed)
> 
> So resolve I would need to revert this commit [8a14cdd](https://github.com/openedx/studio-frontend/pull/341/commits/8a14cdd55f5eb8ee89223c78e86791ce09b3b6ea). And that pacakge shall be removed when `babel-jest` is upgraded to >= 24.* Does that sound okay @Jawayria @aht007
> 
> ## Footnotes
> 1. https://archive.jestjs.io/docs/en/23.x/getting-started.html#using-babel [↩](#user-content-fnref-1-e2b1bc6da4c6808b4eb6bb5149b517ed)

Yeah thats okay with me, we can add this placeholder package until we don't upgrade to babel-jest. We can also mention this thing in comments somewhere to keep it in records to remove that once the said upgrade is done

**Comment by ghassanmas** [comment]
@aht007 can you rerun the tests now... I have readded the bridge pacakge... also wrote expalnation about it in the commit message

**Comment by Jawayria** [comment]
@ghassanmas why can't we just upgrade babel-jest to v24 instead of going for a temporary solution? Is there any blocker on upgrading babel-jest to v24?

**Comment by ghassanmas** [comment]
@Jawayria two tests are failling: 

```
Summary of all failing tests
 FAIL  src/components/EditImageModal/EditImageModal.test.jsx (6.265s)
  ● EditImageModal › Modal › behaves › handleOpenModal sets correct state for event with data (editing an image)

    expect(received).toEqual(expected) // deep equality

    Expected: "normal"
    Received: "skeleton"

      700 |
      701 |         expect(editImageModal.state('areImageDimensionsValid')).toEqual(true);
    > 702 |         expect(editImageModal.state('assetsPageType')).toEqual(pageTypes.NORMAL);
          |                                                        ^
      703 |         expect(editImageModal.state('baseAssetURL')).toEqual(sampleText);
      704 |         expect(editImageModal.state('currentUploadErrorMessage')).toBe(null);
      705 |         expect(editImageModal.state('currentValidationMessages')).toEqual({});

      at Object.<anonymous> (src/components/EditImageModal/EditImageModal.test.jsx:702:56)

 FAIL  src/components/AccessibilityPolicyForm/AccessibilityPolicyForm.test.jsx
  ● <AccessibilityPolicyForm /> › statusAlert › shows correct success message

    expect(received).toContain(expected) // indexOf

    Expected substring: "×Thank you for contacting edX!Thank you for your feedback regarding the accessibility of Studio. We typically respond within one business day (Monday to Friday, 1:00 PM UTC to 9:00 PM UTC)."
    Received string:    "×Thank you for contacting edX!Thank you for your feedback regarding the accessibility of Studio. We typically respond within one business day (Monday to Friday, 3:00 PM GMT+2 to 11:00 PM GMT+2)."

      159 |       expect(statusAlertType).toEqual('success');
      160 |       expect(statusAlert.find('div').first().hasClass('alert-success')).toEqual(true);
    > 161 |       expect(statusAlert.find('div').first().text()).toContain('×Thank you for contacting edX!Thank you for your feedback regarding the accessibility of Studio. We typically respond within one business day (Monday to Friday, 1:00 PM UTC to 9:00 PM UTC).');
          |                                                      ^
      162 |     });
      163 |
      164 |     it('shows correct rate limiting message', () => {

      at Object.<anonymous> (src/components/AccessibilityPolicyForm/AccessibilityPolicyForm.test.jsx:161:54)


Test Suites: 2 failed, 39 passed, 41 total
Tests:       2 failed, 720 passed, 722 total
Snapshots:   0 total
Time:        19.21s      ```
 
 I can fix them but I don't if I would be cheating or not. So I opted out from missing the test cases

**Comment by Jawayria** [comment]
Oh alright. If it brings up test failures then leave the babel-jest upgrade. We can go with the current solution for now. 
Thanks for clarifying the things.

**Comment by aht007** [comment]
@ghassanmas Are we good to merge this one now...?

**Comment by ghassanmas** [comment]
I am good but I can't merge it myself

**Review by aht007 — APPROVED** [review]

**Review by Jawayria — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@ghassanmas 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/studio-frontend&entry.867055334=https://github.com/openedx/studio-frontend/pull/341&entry.1484655318=https://github.com/ghassanmas&entry.752974735=2022-04-28+22:28&entry.1917517419=2022-05-13+10:27&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**